packages/mask/src/extension/popups/pages/Wallet/CollectibleDetail/index.tsx

```diff
@@ -0,0 +1,298 @@
+        color: theme.palette.maskColor.main,
+        marginTop: theme.spacing(3),
+        marginBottom: theme.spacing(1.5),
+    },
+    text: {
+        fontSize: 12,
+        color: theme.palette.maskColor.second,
+        fontWeight: 400,
+        wordWrap: 'break-word',
+    },
+    traits: {
+        flexWrap: 'wrap',
+        gap: theme.spacing(1.5),
+        display: 'grid',
+        gridTemplateColumns: 'repeat(3, 1fr)',
+    },
+    trait: {
+        border: `1px solid ${theme.palette.maskColor.line}`,
+        borderRadius: 8,
+        padding: 6,
+        overflow: 'auto',
+    },
+    traitType: {
+        fontSize: 12,
+        color: theme.palette.maskColor.primaryMain,
+        textTransform: 'capitalize',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+    },
+    traitValue: {
+        fontSize: 13,
+        color: theme.palette.maskColor.primary,
+        marginTop: theme.spacing(0.5),
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+    },
+    sendButton: {
+        position: 'fixed',
+        left: 18,
+        right: 18,
+        bottom: 18,
+        boxShadow: '0px 6px 12px 0px rgba(7, 16, 27, 0.20)',
+        backdropFilter: 'blur(8px)',
+    },
+    iconButton: {
+        padding: 0,
+        minWidth: 'auto',
+        width: 'auto',
+    },
+}))
+
+/**
+ * timestamp in seconds or milliseconds
+ */
+const formatTimestamp = (timestamp: string) => {
+    const value = Number.parseInt(timestamp, 10) * (timestamp.length === 10 ? 1000 : 1)
+    return formatDateTime(new Date(value), 'yyyy-MM-dd HH:mm')
+}
+
+export const CollectibleDetail = memo(function CollectibleDetail() {
+    const { classes } = useStyles()
+    const { t } = useI18N()
+    const navigate = useNavigate()
+    const location = useLocation()
+    const { chainId, address, params } = useTokenParams()
+    const account = useAccount(NetworkPluginID.PLUGIN_EVM)
+    const id = params.get('id') || ''
+    const stateAsset = location.state?.asset as
+        | Web3Helper.NonFungibleTokenScope<void, NetworkPluginID.PLUGIN_EVM>
+        | undefined
+    const { data: asset, isLoading } = useNonFungibleAsset(NetworkPluginID.PLUGIN_EVM, address, id, { chainId })
+
+    useEffect(() => {
+        if (!asset && !isLoading) navigate(-1)
+    }, [!asset && !isLoading, navigate])
+    useTitle(asset?.metadata?.name || t('collectible_title'))
+
+    const availableAsset = asset || stateAsset
+
+    const name = availableAsset?.metadata?.name
+    const collectionName = availableAsset?.collection?.name
+    const assetDesc = availableAsset?.metadata?.description
+    const collectionDesc = availableAsset?.collection?.description
+    const floorPrice = useMemo(() => {
+        if (!asset?.collection?.floorPrices) return null
+        return (
+            asset.collection.floorPrices.find((x) => x.marketplace_id === 'opensea') || asset.collection.floorPrices[0]
```

Comment:
```suggestion
            asset.collection.floorPrices.find((x) => x.marketplace_id === SourceType.OpenSea) || asset.collection.floorPrices[0]
```

Reply:
It's `marketplace`, a little different from `SourceType`